src/components/select-field/BaseSelectField.js

```diff
@@ -386,9 +452,18 @@ class BaseSelectField extends React.Component<Props, State> {
 
     renderSelectOptions = () => {
         const { optionRenderer, options, selectedValues, separatorIndices, shouldShowClearOption } = this.props;
-        const { activeItemIndex } = this.state;
+        const { activeItemIndex, searchText } = this.state;
 
-        const selectOptions = options.map<React.Element<typeof DatalistItem | 'li'>>((item, index) => {
+        let filteredOptions;
+        if (searchText === '') {
```

Comment:
you can somewhat simplify this if you check for `''` in `.filter`. 

`option => !!searchText ? option.displayText.toLowerCase().includes(searchText.toLowerCase()) : true`

Reply:
Good call, I can do that instead.